app/controllers/api/functions.php

```diff
@@ -1471,13 +1472,23 @@
             throw new Exception(Exception::DEPLOYMENT_NOT_FOUND);
         }
 
+        $path = $deployment->getAttribute('path');
+        if(empty($path) || !$deviceForFunctions->exists($path)) {
+            throw new Exception(Exception::DEPLOYMENT_NOT_FOUND);
+        }
+
         $deploymentId = ID::unique();
 
+        $destination = $deviceForFunctions->getPath($deploymentId . '.' . \pathinfo('code.tar.gz', PATHINFO_EXTENSION));
+        $deviceForFunctions->transfer($path, $destination, $deviceForFunctions);
```

Comment:
PR looks okay.. But I need to understand why we use transfer and if it can be done without the overhead of moving files around thereby slowing down the endpoint significantly depending on the size of the deployment

Reply:
We need to copy deployment, so each has different path for their source. Because if they share a source, and then you delete one of the deployments, then the other one doesn't have source (you cannot download, or redeploy anymore)